File: examples/clock.py
#!/usr/local/bin/python3.7

import sys, time

colon = True;

if __name__ == "__main__":
    with open('/dev/tm1637', 'w') as clock:

        """ A nice looking clock starting screen for a second
        """
        print("--:--", file=clock, flush=True)
        time.sleep(1)

        while True:
            try:
                """ Write the current time line once per second
                    w/ and w/out a colon alternately
                """
                if colon:
                    print(time.strftime("%H:%M"), file=clock, flush=True)
                else:
                    print(time.strftime("%H %M"), file=clock, flush=True)
                colon = not colon
                time.sleep(1)

            except KeyboardInterrupt:
                """ Blank the display
                """
                print("", file=clock, flush=True)
                sys.exit()

File: examples/clock.pl
#!/usr/local/bin/perl

use strict;

my $dots = 0;

# Off the display and close device
sub signal_handler {
    print C "\n"; # Trailing spaces will be added
    exit;
};

open C, ">/dev/tm1637" or die "Reason: $!";

# Control+C handler (see above)
$SIG{INT} = \&signal_handler;

# Unbuffered output
select(C);
$|=1;

# A little show off
print C "--:--\n";
sleep(1);

# Main loop
while(1) {
    my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime();
    if ($dots == 0) {
      printf C "%02d %02d\n", $hour, $min;
    }
    else {
      printf C "%02d:%02d\n", $hour, $min;
    }
    $dots = abs(--$dots);
    sleep(1);
}

# Always close the device before exit.
END {
    close C;
}
